### DISCUSSION TITLE
Is there a way to make an attack or anomaly on the CubeSatSim?

### DISCUSSION THREAD

**Discussion by roypyp** [alanbjohnston/CubeSatSim#273] [Q&A]
First, I want to apologize for posting the question in the wrong place.

I wanted to attack satellites to see how it affects the subsystem of the satellite(temperature, power, and more). 
Is it possible to make an anomaly that looks like an attack or send a command(inside Pi Zero because there is no uplink) to create an anomaly in the CubeSatSim? 

I appreciate any help you can provide,
Roy.

**Comment by roypyp** [comment]
Could I use the simulator without the radio components? And get the data from the pi system?

**Reply by alanbjohnston** [reply]
Hi Roy,

The CubeSatSim software runs on a Raspberry Pi running PiOS. So theoretically any attack on this operating system could be used to attack the CubeSatSim and interfere with its operation. 

However, if this exercise is meant to simulate an attack on a real CubeSat, it might not be very useful. This is because few CubeSats run a full Linux OS but instead run their code on a microcontroller such as the STM32 series. So attacks tailored for a microcontroller such as this would be more realistic.

Also, since an attacker is unlikely to have WiFi or physical access to a CubeSat, the attacks will need to come over the radio command and control channel. 

The current CubeSatSim hardware does not have a radio receiver and hence has no radio control channel. Future hardware will have a radio receiver and will have a very simple command interface such as the ability to turn
The radio beacon on and off. 

Does that make sense?

Alan

**Comment by roobixx** [comment]
Roy,

I am currently building out two CubeSats for the exact purposes you have mentioned. As Alan mentioned, the CubeSatSIm does not currently have a radio receiver so I added a RFM95 LoRa module to the STEM board and wired it directly to the GPIO pins on the STEM board for communication with the Pi Zero W via SPI. 

Obviously the CubeSatSim software does not support running this way so I have been writing custom python code that implements a basic command & data handling function that is ripe with vulnerabilities that can be exploited. 

I will be publishing my code/modifications to CubeSatSim in late October that included OpenC3 Cosmos integration and a digital twin (Docker container) for people without hardware.

The work and design that has gone into CubeSatSim is incredible. With a little thought and effort it has proven to be a very capable platform, even if not running it as intended.  Kudus to @alanbjohnston and team.

**Reply by PGpiao** [reply]
Hello, I am a cybersecurity student, I participated in the HACK-A-SAT competition and became interested in it, I am also currently working on the digital twin satellite for other students to practice, but our lack of expertise has caused us to not make good progress. Can you make your project available for students to practice on?
